projects/ion/src/lib/smart-table/smart-table.component.ts

```diff
@@ -61,6 +63,16 @@ export class IonSmartTableComponent implements OnInit, AfterViewChecked {
     }
   }
 
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.config.firstChange) {
+      this.tableUtils = new TableUtils(this.config);
+    }
+
+    if (!changes.config.firstChange) {
+      this.tableUtils.applyPipes(this.config);
+    }
```

Comment:
I think that's the reason why we had 2 calls when we are discussing this solution, `!changes.config.firstChange` is true even if `changes.config` hasn't changed, we can check if this solve our problem:

```suggestion
    if (changes.config) {
      const { firstChange } = changes.config;
      firstChange
        ? (this.tableUtils = new TableUtils(this.config))
        : this.tableUtils.applyPipes(this.config);
    }
```